### PR TITLE
clustering mode

### PR DESCRIPTION
**Pull request by stephenlb**
Now you can launch multiple copies and replicate events across each docker container.  See `readme.md` markdown file for instructions.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,16 +1,18 @@
...
-ADD public /go/src/app/public
-WORKDIR /go/src/app
 RUN go get -d -v
 RUN go install -v
+RUN go build main.go
 
+FROM alpine:3.7
+RUN mkdir -p /go/src/showhook/
+WORKDIR /go/src/showhook/
+COPY --from=0 /go/src/showhook/main .
```
Comment (kaihendry) [inline Dockerfile:13]: what does --from=0 do?

**Comment by kaihendry** [comment]
I don't think this fits the AWS scaling model very well, as you would need to keep a list of your neighbours? Isn't it simpler to leverage a pull model with [SQS long polling](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-long-polling.html)? https://s.natalian.org/2018-05-24/sqs.mp4

Check out my [SQS](https://github.com/unee-t/showhook/tree/sqs) sample code which has a couple of issues at https://github.com/unee-t/showhook/commit/00f74b8b152c91338a97770ae7147fd07a648e66 

* need to wait/block until an eventsource is established before pulling from sqs
* the sqs messageIDs don't correspond to the sns message id... perhaps that's an AWS issue
* in my example, it probably would be better if the message was NOT immediately deleted, just to show multiple clients can receive it. it would be probably better to have a list of what i've seen and then expire messages after 5 minutes or something. Perhaps adjusting the [visibility timeout](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-visibility-timeout.html) might help, but I don't quite understand it.

**Comment by stephenlb** [comment]
Yes this should work nicely as well!  Note that you will have a limit of 100 SQS Topics by default.  You can ask for more.  SQS is great for queuing work and tasks to be processed. 👍

Broadcasting updates with SQS is possible and it wasn't built for it.  SQS is ideal for queuing long-running tasks that need processing time across many EC2 servers.  It can be used as a delivery queue for end-target clients as long as you only have 100 target clients.

File: Dockerfile (hunk 1 above)
Comment (stephenlb) [reply Dockerfile:13]: This allows you to access another container's files.  This way you can have a **`BUILD`** container with build-tools and the golang compiler.  Meanwhile you create a second "`FROM`" **`RUNTIME`** container to host the binary at a fraction of the build container size.

Savings of 400MB down to 12MB.  🎉

Comment (stephenlb) [reply Dockerfile:13]: **`FROM golang:alpine`** is `--from=0` and then **`FROM alpine:3.7`** is `--from=1`

**Comment by kaihendry** [comment]
Ok, I hope you don't mind if I don't accept the PR. Your example is valuable, so please don't delete it. :)

I'm going to modify this to be SQS based.

**Pull request closed without merging**